### PR TITLE
Add member (loyalty) program (#3563)

### PR DESCRIPTION
**Pull request by alex-jansen**

### CODE REVIEW

File: data/ext/pending/issue-3563-examples.txt [hunk 1]
```diff
@@ -0,0 +1,134 @@
+TYPES: #eg-0489 Certification
+
+PRE-MARKUP:
+
```
Comment (MatthiasWiesmann) [inline data/ext/pending/issue-3563-examples.txt:4]: I would not have this generic text, but have instead a textual description of what is represented in the json-ld, i.e. that there a three programs, with the advantages and benefits in plain english.

File: data/ext/pending/issue-3563.ttl [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+:hasTiers a rdf:Property ;
+    rdfs:label "hasTiers" ;
+    :domainIncludes :MemberProgram ;
+    :rangeIncludes :MemberProgramTier ;
+	:inverseOf :isTierOf ;
+    :source <https://github.com/schemaorg/schemaorg/issues/3563> ;
+    rdfs:comment "The tiers of a member program." .
+
+
```
Comment (MatthiasWiesmann) [inline data/ext/pending/issue-3563.ttl:37]: Nitpick, consistent number of empty lines.

File: data/ext/pending/issue-3563.ttl [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+:hasMemberProgram a rdf:Property ;
```
Comment (MatthiasWiesmann) [inline data/ext/pending/issue-3563.ttl:39]: Should there not be a hasMembership property on person to express that John Doe has double platinum membership? This could make sense for social networks, where members might have higher tiers? 

Now if MemberProgramTier was a DefinedTerm, then we could write something like


```JSON

{
 "@context": "https://schema.org/",
 "@type": "Person",
 "name": "John Doe",
 "hasCertification": {
    "@type": "Certification",
    "certificationIdentification": {
        "@type": "MemberProgramTier",
        "name": "blue badge",
        "isTierOf": { 
            "@type": "MemberProgram", 
            "name": "Z Social network tiers",
        }
    }
  }
}
```

File: data/schema.ttl [hunk 4]
```diff
@@ -3779,6 +3783,10 @@ See also the <a href="/docs/hotels.html">dedicated document on the use of schema
     rdfs:label "ReservationPending" ;
     rdfs:comment "The status of a reservation when a request has been sent, but not confirmed." .
 
+:Reserved a :ItemAvailability ;
```
Comment (MatthiasWiesmann) [inline data/schema.ttl:3786]: Move this into another pull request.

Comment (alex-jansen) [reply data/schema.ttl:3786]: ack. same for :MadeToOrder.

File: data/ext/pending/issue-3563.ttl (hunk 2 above)
Comment (alex-jansen) [reply data/ext/pending/issue-3563.ttl:37]: done, also converted tabs to spaces

File: data/ext/pending/issue-3563-examples.txt (hunk 1 above)
Comment (alex-jansen) [reply data/ext/pending/issue-3563-examples.txt:4]: Fixed the TYPES and updated the plain text version under "PRE-MARKUP"

File: data/ext/pending/issue-3563.ttl (hunk 3 above)
Comment (alex-jansen) [reply data/ext/pending/issue-3563.ttl:37]: There already exists a [memberOf](https://schema.org/memberOf) property on [Person](https://schema.org/Person) to represent program membership.  [memberOf](https://schema.org/memberOf) currently has values of type [ProgramMembership](https://schema.org/ProgramMembership). I propose we then expand the range of [memberOf](https://schema.org/memberOf) to also include *MemberProgramTier*, which allows us to model the program and tier the person is a member of? (added to this PR))

**Comment by alex-jansen** [comment]
Addressed comments, PTAL ([demo site](https://1-dot-schema-experiment.uc.r.appspot.com/MemberProgram)).

**Review by MatthiasWiesmann — APPROVED** [review]
